shopinvader/services/customer.py

```diff
@@ -68,37 +68,56 @@ def _validator_sign_in(self):
     def _validator_create(self):
         address = self.component(usage="addresses")
         schema = address._validator_create()
-        exclude_keys = ["type"]
-        for key in exclude_keys:
-            schema.pop(key, None)
         schema.update(
             {
-                "email": {"type": "string", "required": True},
-                "external_id": {"type": "string", "required": True},
-                "vat": {"type": "string", "required": False},
-                "company_name": {"type": "string", "required": False},
-                "function": {"type": "string", "required": False},
+                # Email is mandatory as of today
+                "email": {"type": "string", "required": True, "nullable": False},
             }
         )
+        for key in self._validator_create_exclude_keys():
```

Comment:
IMO, we should declare a dedicated schema for update / create method here and avoid to rely on the schema from address. The code would be more readable, a we would have a clear separation of concern. We could also start using pydantic to define the requests / responses. Models from pydantic will provide a better traceability of where the fieds are declared and used.

Reply:
Yes, I was tempted to change it but for [the reason below](https://github.com/shopinvader/odoo-shopinvader/pull/1194#issuecomment-985417758) I decide to stick to the current approach.